saleor/checkout/utils.py

```diff
@@ -429,15 +429,20 @@ def recalculate_checkout_discount(
     applicable.
     """
     checkout = checkout_info.checkout
-    voucher = get_voucher_for_checkout_info(checkout_info)
-    if voucher is not None:
+    if checkout_info.voucher is not None:
+        voucher = (
+            Voucher.objects.all()
+            .select_for_update()
```

Comment:
I don't think that `select_for_update` is needed here. It will lock the voucher for whole checkout processing.

Reply:
Also, do we need to re-fetch the voucher here? We already have it in checkout_info

Reply:
It was done like that before with a voucher so I reassigned it. If we do not want the voucher to be selected for update `checkout_info.voucher` is enought.